cmd/codegen/util/go.go

```diff
@@ -64,39 +67,112 @@ func EnsureGoPath() error {
 }
 
 // GoGet runs go get to install the specified binary.
-func GoGet(path string, version string, goModules bool) error {
+func GoGet(path string, version string, gopath string, goModules bool, sourceOnly bool) error {
 	modulesMode := "off"
 	if goModules {
 		modulesMode = "on"
 	}
 
 	fullPath := path
 	if version != "" {
-		fullPath = fmt.Sprintf("%s@%s", path, version)
-	}
+		if goModules {
+			fullPath = fmt.Sprintf("%s@%s", path, version)
+		} else {
+			fullPath = fmt.Sprintf("%s/...", path)
+		}
 
+	}
+	args := []string{
+		"get",
+	}
+	if sourceOnly {
+		args = append(args, "-d")
+	}
+	args = append(args, fullPath)
 	goGetCmd := util.Command{
 		Name: "go",
-		Args: []string{
-			"get",
-			fullPath,
-		},
+		Args: args,
 		Env: map[string]string{
 			"GO111MODULE": modulesMode,
+			"GOPATH":      gopath,
 		},
 	}
 	out, err := goGetCmd.RunWithoutRetry()
 	if err != nil {
 		return errors.Wrapf(err, "error running %s, output %s", goGetCmd.String(), out)
 	}
-
+	if !goModules && version != "" {
```

Comment:
Is this branch used?

Reply:
Yes, it's used if something that is being got doesn't support go modules, and a version is specified. For example line 498 of `open_api.go`.